components/tree/nz-tree-node.component.ts

```diff
@@ -275,6 +279,13 @@ export class NzTreeNodeComponent implements OnInit, OnChanges, OnDestroy {
         this.nzTreeNode.isLoading = true;
       }
       this.nzTreeNode.isExpanded = !this.nzTreeNode.isExpanded;
+      const nzTreeNodeChildren = this.nzTreeNode.getChildren();
+      for (const child of nzTreeNodeChildren) {
```

Comment:
I don't think it will work properly when the search value is changed, pay attention to the boundary problem.
Notice:


1.  if any node is matched, its parent node and grandparent node's (etc.) property `canHide` should be reset to false.

2. if the node is not matched, the `canHide` property should be reset to true(including its parent nodes), or it will always not allow to be hidden.